src/gt4py/next/embedded/nd_array_field.py

```diff
@@ -254,3 +253,46 @@ class JaxArrayField(_BaseNdArrayField):
         array_ns: ClassVar[ModuleType] = jnp
 
     common.field.register(jnp.ndarray, JaxArrayField.from_array)
+
+
+def _get_slices_with_named_indices(field: common.Field, named_indices: NamedIndicesOrRanges) -> tuple[slice | None, ...]:
```

Comment:
maybe even better to implement this function as 
```suggestion
def _get_slices_with_named_indices(field: Domain, named_indices: NamedIndicesOrRanges) -> tuple[slice | None, ...]:
```
because we only need domain from field, then it's even more generic (just use `None` for array_ns.newaxis with comment).